olmo/train.py

```diff
@@ -430,7 +432,10 @@ def restore_rng_state(self, rng_state: Dict[str, Any]) -> None:
         random.setstate(rng_state["python"])
         np.random.set_state(rng_state["numpy"])
         torch.set_rng_state(rng_state["torch"])
-        torch.cuda.set_rng_state(rng_state["cuda"])
+        if rng_state["cuda"] is not None:
+            torch.cuda.set_rng_state(rng_state["cuda"])
+        if rng_state["mps"] is not None:
+            torch.mps.set_rng_state(rng_state["mps"])
```

Comment:
This needs to check whether those devices are available. We might be taking a model that was trained on a GPU and try to continue training it on a MPS.

Reply:
If a device is available, but we don't have any random state to set for it, we should print a warning.